domain/application/service/unit.go

```diff
@@ -74,6 +108,34 @@ func (s *Service) AddUnits(ctx context.Context, appName string, units ...AddUnit
 	return nil
 }
 
+func (s *ProviderService) mergeApplicationAndModelConstraints(ctx context.Context, appUUID coreapplication.ID) (coreconstraints.Value, error) {
+	validator, err := s.constraintsValidator(ctx)
+	if err != nil {
+		return coreconstraints.Value{}, internalerrors.Capture(err)
+	}
+	// If the provider doesn't support constraints validation, then we can
+	// just return the zero value.
+	if validator == nil {
+		return coreconstraints.Value{}, nil
+	}
```

Comment:
Is this true? Validating and merging seem like different tasks, so this is a bit confusing

Reply:
So, the naming might seem weird, but that's how it's currently implemented. You'll find in `core/constraints/validator.go` that the Validator does both constraint validation and merge.